src/operations/find.ts

```diff
@@ -89,169 +92,242 @@ export class FindOperation extends CommandOperation<FindOptions, Document> {
+  if (typeof options.timeout === 'boolean') {
+    findCommand.noCursorTimeout = options.timeout;
+  } else if (typeof options.noCursorTimeout === 'boolean') {
+    findCommand.noCursorTimeout = options.noCursorTimeout;
+  }
 
-    if (typeof options.allowPartialResults === 'boolean') {
-      findCommand.allowPartialResults = options.allowPartialResults;
-    }
+  if (typeof options.awaitData === 'boolean') {
+    findCommand.awaitData = options.awaitData;
+  }
 
-    if (options.collation) {
-      if (serverWireVersion < SUPPORTS_WRITE_CONCERN_AND_COLLATION) {
-        callback(
-          new MongoError(
-            `Server ${server.name}, which reports wire version ${serverWireVersion}, does not support collation`
-          )
-        );
+  if (typeof options.allowPartialResults === 'boolean') {
+    findCommand.allowPartialResults = options.allowPartialResults;
+  }
 
-        return;
-      }
+  if (options.collation) {
+    findCommand.collation = options.collation;
+  }
 
-      findCommand.collation = options.collation;
-    }
+  if (typeof options.allowDiskUse === 'boolean') {
+    findCommand.allowDiskUse = options.allowDiskUse;
+  }
 
-    if (typeof options.allowDiskUse === 'boolean') {
-      findCommand.allowDiskUse = options.allowDiskUse;
-    }
+  return findCommand;
+}
 
-    if (this.explain) {
-      // TODO: For now, we need to manually ensure explain is in the options. This will change after cursor refactor.
-      this.options.explain = this.explain.verbosity;
-    }
+function makeLegacyFindCommand(
+  ns: MongoDBNamespace,
+  filter: Document,
+  options: FindOptions
+): Document {
+  const findCommand: Document = {
+    $query: filter
+  };
+
+  if (options.sort) {
+    findCommand.$orderby = formatSort(options.sort);
+  }
 
-    // TODO: use `MongoDBNamespace` through and through
-    server.query(
-      this.ns.toString(),
-      findCommand,
-      { fullResult: !!this.fullResponse, ...this.options, ...this.bsonOptions },
-      (err, result) => {
-        if (err) return callback(err);
-        if (this.explain) {
-          // TODO: NODE-2900
-          if (result.documents && result.documents[0]) {
-            return callback(undefined, result.documents[0]);
-          }
-        }
-
-        callback(undefined, result);
-      }
-    );
+  if (options.hint) {
+    findCommand.$hint = normalizeHintField(options.hint);
+  }
+
+  if (typeof options.returnKey === 'boolean') {
+    findCommand.$returnKey = options.returnKey;
+  }
+
+  if (options.max) {
+    findCommand.$max = options.max;
   }
+
+  if (options.min) {
+    findCommand.$min = options.min;
+  }
+
+  if (typeof options.showRecordId === 'boolean') {
+    findCommand.$showDiskLoc = options.showRecordId;
+  }
+
+  if (options.comment) {
+    findCommand.$comment = options.comment;
+  }
+
+  if (typeof options.maxTimeMS === 'number') {
+    findCommand.$maxTimeMS = options.maxTimeMS;
+  }
+
+  if (options.explain) {
+    findCommand.$explain = true;
+  }
```

Comment:
I think this is overwriting the decorateWithExplain call at the top of the function.

Reply:
Hmm, I don't think so. This is called in a different block of the `execute` method above, where `serverWireVersion < 4`. `decorateWithExplain` it happening in a `serverWireVersion > 4` world, where we use `makeFindCommand` (this is `makeLegacyFindCommand`)

Reply:
Oops yea the diff made it easy to miss

Reply:
Sorry to resurrect, but I think either this needs to change to `findCommand.$explain = 'queryPlanner'` or we need to update [this test](https://github.com/mongodb/node-mongodb-native/blob/master/test/functional/explain.test.js#L562) to differentiate between 3.0 and 3.2+; currently it causes CI to fail.

Reply:
tricky little JS boolean magic :) should be fixed now, PTAL